lib/core/userinfo/components/mocks/authentication.service.mock.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Inject, Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { AlfrescoApiService } from './../../../services/alfresco-api.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthenticationServiceMock {
```

Comment:
Create a new interface that will have the same methods as the real service, and make both the real and the mock service implement this interface:

**Interface:**
![interface](https://user-images.githubusercontent.com/99482959/191744517-805045fb-2725-43c6-a638-88a5c48ae425.png)

**Real service:**
![real](https://user-images.githubusercontent.com/99482959/191744526-d084df6e-3f18-4df9-88de-baf92f20d7a7.png)

**Mocked service:**
![mock](https://user-images.githubusercontent.com/99482959/191744524-e71d798a-e78e-4907-b8a6-5911bebcc924.png)

Reply:
This looks good, but once again I think that in this case it will not work that well. `AuthenticationServiceMock` here has one additional dependency - `MODE` which is not present in original `AuthenticationService`. Also, then I would have to mock a lot of unused functions and these functions/properties which I need are based on `MODE` variable so probably it will not be used in the future by another tests/stories/etc.

Reply:
Okay, let's leave it as is